backend-config/account_association.go

```diff
@@ -29,55 +76,31 @@ func (c *ConfigT) processAccountAssociations() {
 // For each account type, it:
 // 1. Checks if the corresponding account ID exists in the destination config
 // 2. Verifies the account exists in the accounts map
-// 3. Creates an AccountWithDefinition by combining account details with its definition
+// 3. Creates an account by combining account details with its definition
 // 4. Assigns it to the appropriate field in the destination
 //
 // Parameters:
 //   - dest: Pointer to the destination being configured
 func (c *ConfigT) setDestinationAccounts(dest *DestinationT) {
-	// Check and set the regular account if specified in the destination config
+	accountAssociationLogger := pkgLogger.Withn(
+		obskit.DestinationID(dest.ID),
+		obskit.DestinationType(dest.DestinationDefinition.Name),
+	)
+	// Check and set the delivery account if specified in the destination config
 	if accountID, ok := dest.Config["rudderAccountId"].(string); ok {
-		if account, exists := c.Accounts[accountID]; exists {
-			accountDefinition, exists := c.AccountDefinitions[account.AccountDefinitionName]
-			if !exists {
-				pkgLogger.Warnn("Account definition not found in configured accountDefinitions for delivery flow",
-					logger.NewStringField("rudderAccountId", accountID),
-					logger.NewStringField("accountDefinitionName", account.AccountDefinitionName),
-					logger.NewStringField("destinationId", dest.ID))
-			}
-			dest.DeliveryAccount = &AccountWithDefinition{
-				Id:                accountID,
-				Options:           account.Options,
-				Secret:            account.Secret,
-				AccountDefinition: accountDefinition,
-			}
-		} else {
-			pkgLogger.Warnn("Account not found in configured accounts for delivery flow",
-				logger.NewStringField("rudderAccountId", accountID),
-				logger.NewStringField("destinationId", dest.ID))
-		}
+		accountAssociationLogger = accountAssociationLogger.Withn(
+			logger.NewStringField("rudderAccountId", accountID),
+			logger.NewStringField("flowType", "delivery"),
+		)
+		dest.DeliveryAccount = c.populateAccountToDestination(accountID, accountAssociationLogger)
 	}
 
 	// Check and set the delete account if specified in the destination config
 	if deleteAccountID, ok := dest.Config["rudderDeleteAccountId"].(string); ok {
```

Comment:
is an empty check required as well?
```suggestion
	if deleteAccountID, ok := dest.Config["rudderDeleteAccountId"].(string); ok && deleteAccountID != "" {
```